File: Python3/1201 - 1400/1302 Deepest Leaves Sum.py
# 96ms, 67.23%; 16.4MB, 100.00%
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

from collections import deque


class Solution:
    def deepestLeavesSum(self, root: TreeNode) -> int:
        _sum, q = 0, deque([root])
        while q:
            _sum = 0
            for _ in range(len(q)):
                tmp = q.popleft()
                _sum += tmp.val
                if tmp.left: q.append(tmp.left)
                if tmp.right: q.append(tmp.right)
        return _sum

File: 1201 - 1400/1302 Deepest Leaves Sum.cc
// 52ms, 52.98%; 32.1MB, 100.00%
#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    int deepestLeavesSum(TreeNode *root) {
        int _sum = 0;
        queue<TreeNode *> q;
        q.push(root);
        while (!q.empty()) {
            _sum = 0;
            for (int i = q.size(); i > 0; --i) {
                auto tmp = q.front();
                q.pop();
                _sum += tmp->val;
                if (tmp->left) q.push(tmp->left);
                if (tmp->right) q.push(tmp->right);
            }
        }
        return _sum;
    }
};
